EVG-17305 Don't activate builds during addNewTasks if no tasks are active

malikchaya2 opened the pull request:
[EVG-17305](https://jira.mongodb.org/browse/EVG-17305)

### Description 
When generating tasks, if the build already exists, the addNewTasks function activates the build even if it shouldn't be activated. 

### Testing 
Tested on staging. [Before](https://spruce-staging.corp.mongodb.com/version/638eb4e79dbe327e7b1b0be1/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC) and [After](https://spruce-staging.corp.mongodb.com/version/638eb79197b1d331945c27e4/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC).

Kimchelly reviewed (commented):
Is there a test case for this case when all the tasks in the build variant are inactive? If not, can we add one? Is it also worth testing the explicitly setting `activate: false` for the build variant keeps it deactivated (or is that already tested)?

Kimchelly commented:
I still have the same question about the tests.

malikchaya2 commented:
> I still have the same question about the tests.

I added a case to test this explicitly, and added a check for the bv activated state to each case.

Kimchelly commented on model/lifecycle_test.go:
```diff
@@ -2587,10 +2607,13 @@ func TestAddNewTasks(t *testing.T) {
 			activatedTasks, err := task.FindAll(db.Query(bson.M{task.ActivatedKey: true}))
 			assert.NoError(t, err)
+			build, err := build.FindOneId("b0")
+			assert.NoError(t, err)
```
Is this needed? Seems like we don't need to double-check for the existence of `b0`.

malikchaya2 replied:
We need to pull it from the db to check if addNewTasks activated the build.

Kimchelly replied:
Ah I see. Does it need to also check if `build` is non-nil? Just checking the error won't check for the doc not found case.

Kimchelly replied:
The test will panic on that line if `build` is nil, but I kind of prefer that tests don't panic if they don't have to (because they stop the entire test runtime, rather than just this test). I don't think this is a hard requirement given how often the existing tests _can_ panic, but it is part of the style guide and it would be nice if we could aim for less panicking in tests.

malikchaya2 replied:
good point, fixed

Kimchelly reviewed (approved):
The test looks good! LGTM mod one question about the test.
